Nanotrasen Prometheus-Class Plasma Tanker

BBBBAface opened the pull request:
## About The Pull Request

Shiptest didn't want it so here is a ship I spent about 20 hours working on. There weren't any ships like it on Shiptest; I'm not sure if there are any plasma tankers on this codebase or not, but here's the fully functioning ship.


This PR adds a new Nanotrasen ship to the game, A plasma transport vessel with limited facilities but enough plasma to last a large settlement indefinitely.

![2022 09 11-15 03 59](https://user-images.githubusercontent.com/19318747/189544734-f0a58d99-d56e-4d09-ae44-e28f979d6b63.png)
![2022 09 11-15 04 08](https://user-images.githubusercontent.com/19318747/189544736-593e40fb-b144-476e-b559-53e39dc8110c.png)


Specs:
Limited to 2 per round
440Ton weight with 6 ion engines
Supports a crew of up to 7, but works best with 3 - 5 Power-management system 
Around 2,500,000 MOL of plasma gas
5,000cr starting cash
Stocked medbay
Extremely limited weaponry
Basic mining gear
Working Exterior / Engine / Bay shutters and holofans
"Robust" atmos setup
Outpost comms console

Pros: 
• You will never run out of plasma gas
• Defribulator and surgical tools
• A cell for troublesome patients
• Engineering protolathe
• Large bay doors and engineering area
• Easily customizable
• Only ship of its kind
• Slot machine!
• Shoilet!

Cons:
• NO R&D capabilities
• Limited security gear
• NO way of actually utilizing the plasma it has stored
• Only 3 hardsuits
• No launchpad or outpost trading console
• Slightly cramped design
• No insulated gloves; you're going to have to drink Rubbernecks




- [X] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game

Currently, no ships are designed to trade gas, leaving the niche unfilled. This is the solution.

This ship would be a decent trading vessel, and because of the defibrillator, it could also make a good emergency response vessel, able to refuel ships and revive the crew. Lack of plasma engines or TEG means the plasma gas is useless on round start. Trading with other players is required due to the lack of research capabilities.

## Changelog

:cl:
Add: Prometheus-class tanker
/:cl:

BBBBAface committed:
Added whole complete project

shiptest didn't want it so

MoonFalcon commented:
Looks like it's failing linting. Really like the ship by the way

BBBBAface committed:
fixes errors

if not idk, feel free to fix them if you want to use the ship because i don't play voidcrew and apparently the codebase is different enough for many things to not work / be here

BBBBAface committed:
i lied theyre all fixed now

BBBBAface committed:
fr

PeriodicChaos commented:
Nice ship i hope to see it ingame no idea why shiptest would want to give this up.

PeriodicChaos commented:
heres the error something about an id 
![image](https://user-images.githubusercontent.com/44913068/188989136-015ddcf4-3998-46e4-a56e-4a4a40088dd5.png)

Jackriip commented:
Looks great! I'd be happy to test merge it if you get the linting fixed

BBBBAface commented:
> Nice ship i hope to see it ingame no idea why shiptest would want to give this up.

they said to wait for a rework to "the gas economy" and redo the pr when that's done, so basically just wait indefinitely.

BBBBAface committed:
frfr

GuiltyNeko commented on _maps/Configs/nanotrasen_tanker.json:
```diff
@@ -0,0 +1,33 @@
+{
+	"$schema": "https://raw.githubusercontent.com/shiptest-ss13/Shiptest/master/_maps/ship_config_schema.json",
+	"prefix": "NTSV",
```
This will need to be "NEU", as, tragically, we only have three accepted prefixes here.

GuiltyNeko reviewed (changes requested):
Window spawners spawn grilles under them, you should remove the grilles on the same tiles as your window spawners. In addition, you should put the ID back on the door, the linters just didn't like the ID var on the docking port. I'd also like you to replace the bridge holopad with a holopad/emergency/command, as most/all of our ships have one and that one in particular can help with access issues faced by a crew. And if I remember correctly, cyclelink helpers only work on doors in cardinal directions from each other. If you want the airlock to cycle in that way you'll need cyclelink_helper_target.

BBBBAface committed:
fixes prefix

BBBBAface commented:
> Window spawners spawn grilles under them, you should remove the grilles on the same tiles as your window spawners. In addition, you should put the ID back on the door, the linters just didn't like the ID var on the docking port. I'd also like you to replace the bridge holopad with a holopad/emergency/command, as most/all of our ships have one and that one in particular can help with access issues faced by a crew. And if I remember correctly, cyclelink helpers only work on doors in cardinal directions from each other. If you want the airlock to cycle in that way you'll need cyclelink_helper_target.

The airlock is fully functional as-is, this ship has been playtested quite a bit. I'm fixing the issues and updating the branch.

BBBBAface committed:
deletes grilles and swaps holopad

airlocks work good (thumbs up emoji)

BBBBAface commented:
I put the id back on the door and it returns an error again, should I just delete it?

GuiltyNeko commented:
Ah, on doors, the proper var name is id_tag. That should do it.

BBBBAface committed:
fixes door tag

all good now

GuiltyNeko reviewed (approved):
I'll TM this on the next round of testmerges

BBBBAface committed:
made price reasonable

BBBBAface committed:
Merge branch 'voidcrew:master' into master

BBBBAface committed:
reorganizes bridge and slightly un-nerfs sec

I looked at other ships in this codebase and realized the near complete lack of security gear would likely discourage people from picking this ship. Hopefully it's good in it's current state.

BBBBAface committed:
Merge branch 'voidcrew:master' into master

BBBBAface committed:
this is the file i shouldve uploaded

BBBBAface committed:
Tweaks to floors

Also adds turbine boards (@Bread Man#7984's idea)

BBBBAface committed:
Delete nanotrasen_tanker.json

BBBBAface committed:
Delete nanotrasen_tanker.dmm

BBBBAface committed:
changes NEU to NT-C

i just made it neu without looking at what it stood for lmao

BBBBAface committed:
minor tweaks

BBBBAface committed:
Delete NEU_tanker.json

BBBBAface committed:
changes output on tanks to manual valve

Making the tank connectors instead of open-ended manual valves is something the shiptest devs wanted me to change, even though existing ships have open-ended valves already. Tank connectors on the end of the pipe makes no sense whatsoever and i forgot to change it back. This one really irritated me, no clue why they would make me wait for 2 days telling me to change that and then say they didn't even want the ship

BBBBAface committed:
Merge branch 'voidcrew:master' into master

BBBBAface committed:
fuck it, its neutral

a bad frag nade and the ship is gone, so

BBBBAface committed:
fixes floor pipes

BBBBAface committed:
fixes wires too

BBBBAface committed:
Merge branch 'voidcrew:master' into master

BBBBAface committed:
Rename NT-C_tanker.json to NEU_tanker.json

BBBBAface committed:
Update turf_transparency.dm

BBBBAface committed:
Update _material.dm

BBBBAface committed:
Update change_turf.dm

BBBBAface committed:
Update misc_floor.dm

BBBBAface committed:
Update glass.dm

BBBBAface committed:
Update tile_types.dm

BBBBAface committed:
Update glass.dm

BBBBAface committed:
Update glass.dm

BBBBAface committed:
icons update

BBBBAface committed:
Merge remote-tracking branch 'origin/Floor-fixes-galore'

BBBBAface committed:
Merge branch 'master' of https://github.com/BBBBAface/Voidcrew-LRP-ships

BBBBAface commented:
fixing my GitHub stuffi messed up awhile back, im just going to redo this pr

BBBBAface committed:
Revert "Merge branch 'master' of https://github.com/BBBBAface/Voidcrew-LRP-ships"

This reverts commit b0ffa72efecfb1e0d57f2872ada76ee227282e8c, reversing
changes made to c39e3bcbd194617d47d53a12a7649762d14c1bfe.

BBBBAface committed:
Revert "Merge remote-tracking branch 'origin/Floor-fixes-galore'"

This reverts commit c39e3bcbd194617d47d53a12a7649762d14c1bfe, reversing
changes made to 48d6a976cc21b09f53b22640271e2963309934b8.
